### ISSUE TITLE
How to generate a grid from a rectangle with fixed size for the hexagon?

### ISSUE THREAD

**Issue by khamaileon** [flauwekeul/honeycomb#31]
I need to generate a hexagonal grid in my JS frontend that perfectly matches the one I generate from my Python backend because I use the centroid of the hexagon as an ID.

This is the Python code / library I use now:

https://github.com/stephanh42/hexutil#converting-between-hexagonal-grid-coordinates-and-screen-coordinates

```python
hex_width = 100
hex_height = 58
grid = HexGrid(width=hex_width, height=hex_height)
hexes = grid.hexes_in_rectangle([x_min, y_min, width, height])
```

Is there a way to fix the height and width of a hexagon with "honeycomb"?

**Comment by flauwekeul** [comment]
If I understand you correctly you want to have a different width and height for your hexes. I'm afraid this is not possible currently with honeycomb ☹️ 

I can image this is an important feature and I'll look into adding it. Please keep in mind this might take some time though...

**Comment by flauwekeul** [comment]
Update: I've added the possibility to set a separate width and height for hexes, but `Hex#fromPoint()` (and `Grid.pointToHex()`) is now broken. I'm struggling with the math, but once I've got it, I need to update the tests and I could release an alpha version. Hopefully within a week.

**Comment by redblobgames** [comment]
One way to think about `Hex#fromPoint` is that you now have a separate `size` for x and y axes. Whereas before, you might calculate hex width and height from `size`, you now have them as input constraints, and can calculate `size` from them (separately for x and y).

**Comment by flauwekeul** [comment]
Thanks for the input Amit :blush: I know how I'm going to do it, didn't found the time to finish it though :pensive:

**Comment by flauwekeul** [comment]
I've just released [v2.0.0](https://github.com/flauwekeul/honeycomb/blob/master/CHANGELOG.md#200-2019-05-08) of honeycomb. It's a major release because I've removed some methods. I hoped the auto generated changelog would show the breaking changes too, but no 😞

In this new version it's possible to have "stretched" hexes. See the [docs](https://github.com/flauwekeul/honeycomb#size) for more info.

I've actually used a (the?) [python hex library](https://github.com/gojuno/hexgrid-py) to "borrow" the math, thanks for that 👍
